### DISCUSSION TITLE
Expandable rows feature

### DISCUSSION THREAD

**Discussion by RROSANN** [vmware/clarity#6353] [Q&A]
Can you detect if the rows are open ? I made a custom button that will open all of them but does if all of them are open it will not detect them so to say... Example if I open the manually could I detect if all of them are open and use that to change my custom button? 
![type1](https://user-images.githubusercontent.com/74143411/134299880-ffec2588-f96d-49c0-8fd1-1ca273a3f16f.png)
![type2](https://user-images.githubusercontent.com/74143411/134299882-a9e28650-9f53-487c-8548-5045cb6ef691.png)
![type3](https://user-images.githubusercontent.com/74143411/134299888-39fb7b7f-0eda-4538-92e9-251793e2882e.png)

**Comment by hippee-lee** [comment]
One approach might be to de-sugar the `*clrIfExpanded` into 

```html 
<clr-dg-row-detail [clrIfExpanded]="expandedRowIDValue" (clrIfExpandedChange)="handleExpandedRowChange($event, expandedRowID)">
```

This would allow the application to manage the expanded state for rows either all at once or for individual rows.
